extension/js/first-run.js

```diff
@@ -0,0 +1,34 @@
+document.addEventListener("DOMContentLoaded", async () => {
+  const { relaySiteOrigin } = await browser.storage.local.get("relaySiteOrigin");
+  const { fxaOauthFlow } = await browser.storage.local.get("fxaOauthFlow");
+
+  const signInBtn = document.getElementById("first-run-sign-in");
+  const joinWaitlistLink = document.getElementById("first-run-join-waitlist-link");
+
+  const appendParams = (url, params) => {
+    for (const utmKey in params) {
+      url.searchParams.append(utmKey, params[utmKey]);
+    }
+    return url;
+  };
+
+  joinWaitlistLink.addEventListener("click", (e) => {
+    e.preventDefault();
+    const utms = {
+      "utm_source": "fx-relay-addon",
+      "utm_medium": "first-run-landing-page",
+      "utm_campaign": "beta",
+      "utm_content": "join-waitlist",
+    };
+
+    let relayUrl = new URL(relaySiteOrigin);
+    relayUrl = appendParams(relayUrl, utms);
+    return window.open(relayUrl.href);
+  });
+
+  // TODO Add FXA params, do metrics flow from extension?
```

Comment:
👍 to add utm params. I thought we figured out that they *do* propagate if you encode them and put them into `auth_params`? e.g.:

`const openFxaFlowWithUtms = openFxaFlow + '&auth_params' + encodeURI('utm_source=fxa-relay-addon&utm_medium=first-run-landing-page&utm_campaign=beta&utm_content=sign-in')`

Reply:
They do _but_ I believe we'll need the beginning `flowId` and `flowBeginTime` are required or else nothing registers.